mailbox/api/src/main/java/org/apache/james/mailbox/model/FetchGroupImpl.java

```diff
@@ -29,27 +29,23 @@
  */
 public class FetchGroupImpl implements MessageResult.FetchGroup {
 
-    public static final MessageResult.FetchGroup MINIMAL = new FetchGroupImpl(MessageResult.FetchGroup.MINIMAL);
+    public static final FetchGroupImpl MINIMAL = new FetchGroupImpl(MessageResult.FetchGroup.MINIMAL);
```

Comment:
why would you want to promote FetchGroupImpl in place of FetchGroup?

Reply:
I believe that arguments of our api should not be interfaces, especially when we expect the caller to insranciate it.

I can not recall how many time i asked myself: how to build a fetch group?

And I don t see the value of having fetchGroup and fetchgroupimpl being separate entities.

Reply:
Either we decide there's no need for the interface at all or we need to pass the interface around, not the implementation.

Of course, in this case I don't thing there's any value in having an interface

Reply:
For API parameters, I'm clearly in favor of having POJO.